internal/resources/ekscluster/conversions.go

```diff
@@ -0,0 +1,410 @@
+package ekscluster
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	eksmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/ekscluster"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func constructSpec(d *schema.ResourceData) (spec *eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec) {
+	spec = &eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec{
+		ClusterGroupName: clusterGroupDefaultValue,
+	}
+
+	value, ok := d.GetOk(specKey)
+	if !ok {
+		return spec
+	}
+
+	data := value.([]interface{})
```

Comment:
all the explicit type conversions should be read as 
data,_ := value.([]interface{}) to avoid runtime crashes

Reply:
I kept this intentionally because any such runtime crashes will be easily caught during dev testing. But, if we change the schema and don't update the construct functions, then we won't be able to catch it. No hard preferences here though

Reply:
No, runtime crashes is looked down up on the provider and would reduce the confidence of the customer using it.
Such changes in the schema needs to be tested and validate via logs , acceptance test or manual testing.

Can you please ensure that the provider does not panic and crash

Reply:
I'll make this change- just as a heads up, it looks like it's been done this way (without an explicit check) in three other places. Might be worth looking into later!

![image](https://user-images.githubusercontent.com/1689056/203403203-92600433-f351-4274-a4a3-72f025dc41f0.png)

Reply:
Fixed